README.md

```diff
@@ -73,24 +56,221 @@ The `-v /path/to/your/local/models:/workspace/models` part will make sure
+in `data_classes.py`.
+
+For examples on how to create spacer tasks, refer to the Core API section below, and the unit tests in `test_tasks.py`.
+
+Tasks can be executed directly by calling the methods in tasks.py. 
+However, spacer also supports an interface with AWS Batch 
+handled by `env_job()` in `mailman.py`. 
+
+Spacer supports four storage types: `s3`, `filesystem`, `memory` and `url`.
+ Refer to `storage.py` for details. The memory storage is mostly used for 
+ testing, and the url storage is read only.
+
+`config.py` defines configurable variables/settings and various constants.
+
+
+## Core API
+
+The `tasks.py` module has four functions which comprise the main interface of pyspacer:
+
+### extract_features
+
+Takes an image, and a list of pixel locations on that image. Produces a single feature vector out of the image-data at those pixel locations. Example:
+
+```python
+from spacer.messages import DataLocation, ExtractFeaturesMsg
+from spacer.tasks import extract_features
+
+message = ExtractFeaturesMsg(
+    # Identifier to set this extraction job apart from others. Makes sense
+    # to use something that uniquely identifies the image.
+    job_token='image1',
+    # Extractors available:
+    # 1. 'efficientnet_b0_ver1': Generally recommended
+    # 2. 'vgg16_coralnet_ver1': Legacy, requires Caffe
+    # 3. 'dummy': Produces feature vectors which are in the correct format but
+    # don't have meaningful data. The fastest extractor; can help for testing.
+    feature_extractor_name='efficientnet_b0_ver1',
+    # (row, column) tuples specifying pixel locations in the image.
+    # Note that row is y, column is x.
+    rowcols=[(2200, 1000), (1400, 1500), (3000, 450)],
+    # Where the input image should be read from.
+    image_loc=DataLocation('filesystem', '/path/to/image1.jpg'),
+    # Where the feature vector should be output to.
+    # CoralNet uses a custom .featurevector extension for these, but the
+    # format is just JSON.
+    feature_loc=DataLocation('filesystem', '/path/to/image1.featurevector'),
+)
+return_message = extract_features(message)
+print("Feature vector stored at: /path/to/image1.featurevector")
+print(f"Runtime: {return_message.runtime}")
+```
+
+### train_classifier
+
+Takes:
+
+- Feature vectors, each vector corresponding to a set of pixel locations in one image
+- Ground-truth (typically human-confirmed) annotations corresponding to those feature vectors
+- Optionally, previously-created classifiers to re-evaluate with these annotations
+- Training parameters
+
+Produces a classifier (model) loadable in scikit-learn, and classifier evaluation results. Example:
+
+```python
+from spacer.data_classes import ImageLabels
+from spacer.messages import DataLocation, TrainClassifierMsg
+from spacer.tasks import train_classifier
+
+message = TrainClassifierMsg(
+    # Identifier to set this training job apart from others. Should be
+    # unique for each classifier.
+    job_token='classifier1',
```

Comment:
Does the uniqueness apply across all jobs (and does that assume a certain `Batch` config?)? Or literally just "for each classifier"?

Reply:
Oh. I'm glad you pointed this out because yes, it's purely bookkeeping for the caller. For the CoralNet + AWS Batch workflow, the token incidentally only has to be unique across other train-classifier jobs (not across all jobs). But that shouldn't be specified in this comment as if it were a pyspacer-level requirement.

Reply:
See commit be57ec7